src/components/common/AddressBookInput/index.tsx

```diff
@@ -13,37 +15,67 @@ const abFilterOptions = createFilterOptions({
 /**
  *  Temporary component until revamped safe components are done
  */
-const AddressBookInput = ({ name, ...props }: AddressInputProps): ReactElement => {
+const AddressBookInput = ({ name, canAdd, ...props }: AddressInputProps): ReactElement => {
   const addressBook = useAddressBook()
   const { setValue, control } = useFormContext()
   const addressValue = useWatch({ name, control })
+  const [open, setOpen] = useState(false)
 
   const addressBookEntries = Object.entries(addressBook).map(([address, name]) => ({
     label: address,
     name,
   }))
 
+  const hasVisibleOptions = useMemo(
+    () => !!addressBookEntries.filter((entry) => entry.label.includes(addressValue)).length,
+    [addressBookEntries, addressValue],
+  )
+
+  const handleOpenAutocomplete = () => {
+    setOpen((value) => !value)
+  }
+
   return (
-    <Autocomplete
-      value={addressValue || ''}
-      disabled={props.disabled}
-      readOnly={props.InputProps?.readOnly}
-      freeSolo
-      options={addressBookEntries}
-      onInputChange={(_, value) => setValue(name, value, { shouldValidate: true })}
-      filterOptions={abFilterOptions}
-      componentsProps={{
-        paper: {
-          elevation: 2,
-        },
-      }}
-      renderOption={(props, option) => (
-        <Typography component="li" variant="body2" {...props}>
-          <EthHashInfo address={option.label} name={option.name} shortAddress={false} />
+    <>
+      <Autocomplete
+        open={open}
+        onOpen={() => setOpen(true)}
+        onClose={() => setOpen(false)}
```

Comment:
Why does this now need to be controlled?

Reply:
So I needed to make this controlled to pass the handler to the `AddressInput` to control the Autocomplete opening from the endAdornment icon (on top of just clicking the input).